gcp/src/test/java/org/apache/iceberg/gcp/biglake/BigLakeCatalogTest.java

```diff
@@ -0,0 +1,385 @@
+  public void before() throws Exception {
+    mockMetastoreService = new MockMetastoreService();
+    mockServiceHelper =
+        new MockServiceHelper(
+            UUID.randomUUID().toString(), Arrays.<MockGrpcService>asList(mockMetastoreService));
+    mockServiceHelper.start();
+
+    File warehouse = temp.toFile();
+    warehouseLocation = warehouse.getAbsolutePath();
+
+    ImmutableMap<String, String> properties =
+        ImmutableMap.of(
+            GCPProperties.BIGLAKE_PROJECT_ID,
+            GCP_PROJECT,
+            CatalogProperties.WAREHOUSE_LOCATION,
+            warehouseLocation);
+
+    channelProvider = mockServiceHelper.createChannelProvider();
+    MetastoreServiceSettings settings =
+        MetastoreServiceSettings.newBuilder()
+            .setTransportChannelProvider(channelProvider)
+            .setCredentialsProvider(NoCredentialsProvider.create())
+            .build();
+
+    bigLakeCatalogUsingMockService = new BigLakeCatalog();
+    bigLakeCatalogUsingMockService.setConf(new Configuration());
+    bigLakeCatalogUsingMockService.initialize(
+        CATALOG_ID,
+        properties,
+        GCP_PROJECT,
+        GCP_REGION,
+        new BigLakeClient(settings, GCP_PROJECT, GCP_REGION));
+
+    mockBigLakeClient = mock(BigLakeClient.class);
+    bigLakeCatalogUsingMockClient = new BigLakeCatalog();
+    bigLakeCatalogUsingMockClient.setConf(new Configuration());
+    bigLakeCatalogUsingMockClient.initialize(
+        CATALOG_ID, properties, GCP_PROJECT, GCP_REGION, mockBigLakeClient);
+  }
+
+  @AfterEach
+  public void after() {
+    mockServiceHelper.stop();
+  }
+
+  @Override
+  protected boolean requiresNamespaceCreate() {
+    return true;
+  }
+
+  @Override
+  protected BigLakeCatalog catalog() {
+    return bigLakeCatalogUsingMockService;
+  }
+
+  @Override
+  protected boolean supportsNamesWithSlashes() {
+    return false;
+  }
+
+  @Test
+  public void testDefaultWarehouseWithDatabaseLocation_asExpected() {
+    when(mockBigLakeClient.getDatabase(DatabaseName.of(GCP_PROJECT, GCP_REGION, CATALOG_ID, "db")))
+        .thenReturn(
+            Database.newBuilder()
+                .setHiveOptions(HiveDatabaseOptions.newBuilder().setLocationUri("db_folder"))
+                .build());
+
+    assertEquals(
+        "db_folder/table",
+        bigLakeCatalogUsingMockClient.defaultWarehouseLocation(TableIdentifier.of("db", "table")));
+  }
+
+  @Test
+  public void testDefaultWarehouseWithoutDatabaseLocation_asExpected() {
+    when(mockBigLakeClient.getDatabase(DatabaseName.of(GCP_PROJECT, "us", CATALOG_ID, "db")))
+        .thenReturn(
+            Database.newBuilder().setHiveOptions(HiveDatabaseOptions.getDefaultInstance()).build());
+
+    assertEquals(
+        warehouseLocation + "/db.db/table",
+        bigLakeCatalogUsingMockClient.defaultWarehouseLocation(TableIdentifier.of("db", "table")));
+  }
+
+  @Test
+  public void testRenameTable_differentDatabase_fail() {
+    Exception exception =
```

Comment:
please use AssertJ-style assertions for this. This would be something like: `assertThatThrownBy(() -> client.renameTable(...)).isInstanceOf(IllegalArgumentException.class).hasMessage(...)`.

Reply:
Done, replaced all.